nilearn/_utils/niimg_conversions.py

```diff
@@ -178,17 +178,12 @@ def iter_check_niimg(
                     )
                 else:
                     raise ValueError(
-                        "Field of view of image #%d is different from "
+                        f"Field of view of image #{i} is different from "
                         "reference FOV.\n"
-                        "Reference affine:\n%r\nImage affine:\n%r\n"
-                        "Reference shape:\n%r\nImage shape:\n%r\n"
-                        % (
-                            i,
-                            ref_fov[0],
-                            niimg.affine,
-                            ref_fov[1],
-                            niimg.shape,
-                        )
+                        f"Reference affine:\n{ref_fov[0]:r}\n"
```

Comment:
seems this line is causing several tests failures

I remember that when I was converting old string formats (% and .format) to f strings, some would lead to some tests failures if I did. I think this may have been one of them.
Sorry for not flagging it in a comment.

If you can find how to fix and keep the f strings, that'd be great, otherwise we should ignore this fix by ruff here.

Reply:
`:r` → `!r`